### ISSUE TITLE
numerical error with the use of range

### ISSUE THREAD

**Issue by orangeluc** [NetLogo/NetLogo#1607]
I try to use "range" command to generate a series of numbers in NetLogo version 6.0.3 (Mac).
When I enter 
        `show (range 3 2.3 -0.1)`
I am expecting `[3 2.9 2.8 2.7 2.6 2.5 2.4]`
However, I get `[3 2.9 2.8 2.7 2.6 2.5 2.4 2.3]`

**Comment by LaCuneta** [comment]
@orangeluc Thank you for reporting the issue, we appreciate it!  

This is a [known issue #1575](https://github.com/NetLogo/NetLogo/issues/1575), and may be addressed in the future.  It's due to how floating point numbers work, and slight amounts of error causing `2.3` to be included (it's probably some value like `2.300000000001` when calculated).  See the "Floating point accuracy" section of the [NetLogo manual](http://ccl.northwestern.edu/netlogo/docs/programming.html#math) for more.  

One workaround (there is another suggested in the other issue) is to use whole numbers with `range` to avoid the floating point issues and then divide them when you're ready to use:  `show (map [[x] -> x / 10] (range 30 23 -1))`.

**Comment by qiemem** [comment]
This is actually a different issue. #1575 is more of a feature request and is (pretty much) inevitably a result of floating point arithmetic. This is also a result of floating point arithmetic, but we already take steps to mitigate issues like this. I thought we had stamped them out, but I guess not!

Note issues like this are not unique to netlogo. Consider numpy's `arange`:

```
In [8]: np.arange(1./3., 1., 1./3.)
Out[8]: array([0.33333333, 0.66666667, 1.        ])
```

We get around stuff like this by having a very small epsilon that we allow as an error margin (1 ulp). However, this issue shows that it is too small! The fix should just be to increase it a little.
